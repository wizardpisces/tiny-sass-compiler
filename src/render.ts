import fs from 'fs'
import baseCompile from './compile'
import { CompilerOptions, CodegenResult } from './type'

export type RenderOptions = {
    filename: string
}

export type RenderCallback = (err: null | Error, result: CodegenResult | null) => void

export function requireCSS(scssPath: string): CompilerOptions {
    return {
        source: fs.readFileSync(scssPath, 'utf8'),
        filename: scssPath
    }
}

export function render(options: RenderOptions, cb: RenderCallback) {
    try {
        let cssRequired = requireCSS(options.filename),
            result = baseCompile(cssRequired.source, cssRequired)
        cb(null, result)
    } catch (e) {
        cb(e, null)
    }
}

export function renderSync(options:RenderOptions){
    let cssRequired = requireCSS(options.filename),
        result = baseCompile(cssRequired.source, cssRequired)
    return result
}