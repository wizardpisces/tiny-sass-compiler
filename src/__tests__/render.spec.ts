import { render, renderSync } from '../render'

describe('render: tests', () => {
    test('render', () => {
        render({ filename:__dirname+'/scss/basic.scss'},(err,res)=>{
            if(res){
                expect(res.code).toMatchSnapshot()
            }
        })
    })
    test('renderSync', () => {
        const res = renderSync({ filename: __dirname+'/scss/basic.scss'})
        expect(res.code).toMatchSnapshot()
    })
})