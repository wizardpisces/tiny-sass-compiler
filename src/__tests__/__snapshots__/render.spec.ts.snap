// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render: tests render 1`] = `
"body .test{
  font:100% Helvetica, sans-serif;
  color:#333;
}
"
`;

exports[`render: tests renderSync 1`] = `
"body .test{
  font:100% Helvetica, sans-serif;
  color:#333;
}
"
`;
